blocks/library/embed/index.js

```diff
@@ -114,6 +121,10 @@ function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, k
 							return;
 						}
 						response.json().then( ( obj ) => {
+							if ( obj.html && -1 !== obj.html.indexOf( 'class="wp-embedded-content"' ) ) {
+								obj.type = 'wp-embed';
```

Comment:
The mutation of `obj` here makes me a bit uneasy, especially since we pass the object around (see below `this.getPhotoHtml( obj )`). I might instead suggest making this part of the destructuring and reassignment of `type`:

```js
let { html, type } = obj;
if ( /* ... */ ) {
	type = 'wp-embed';
}
```

Problem here is that you might run into a `prefer-const` ESLint warning since `html` is not reassigned. Generally speaking, I'd be in favor of enabling the [`"destructuring": "all"` option](https://eslint.org/docs/rules/prefer-const#destructuring), but otherwise you might either have to avoid the destructuring for `html` or destructure separately:

```js
const { html } = obj;
let { type } = obj;
```

Reply:
Ok, i understand your concern, i'll try to "destructure separately".